crates/yulgen/src/types.rs

```diff
@@ -175,22 +173,41 @@ impl AbiType {
     }
 }
 
-impl From<&FixedSize> for AbiType {
-    fn from(typ: &FixedSize) -> Self {
-        match typ {
-            FixedSize::Base(base) => base.into(),
-            FixedSize::Array(array) => array.into(),
-            FixedSize::Tuple(tuple) => tuple.into(),
-            FixedSize::String(string) => string.into(),
+impl EvmSized for AbiType {
```

Comment:
I see that this implemented solely for the construction of struct runtime functions and read the note about not wanting to pass the `db` everywhere.

It is a bit confusing, as this doesn't really reflect the size of the `AbiType`. For example, the size of `uint16[5]` would give *2 \* 5 -> 10*, when the size of the encoded array is actually *32 \* 5 -> 160*. Furthermore, the construction of these struct functions shouldn't be concerned with the ABI, since they only deal with our internal representation of structs. With this being the case, I don't think we should be passing `AbiType`s into the `struct_apis` function. Instead, we should go back to passing in `FixedSize` values.

I think the other places that have been refactored to use `AbiType`s are better now, though.

Reply:
Yeah, good point. `struct_apis` now takes struct fields as `&[(String, FixedSize)]`. (Still avoiding passing in the db here, because that would make testing a pain.)